### PR TITLE
Gjoranv/allow duplicate bundles

### PR DESCRIPTION
**Pull request by gjoranv**
Take a look, but do not merge. I have yet to get proof of concept in a proper test.

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/container/core/config/BundleLoader.java [hunk 1]
```diff
@@ -47,16 +53,19 @@ public BundleLoader(Osgi osgi) {
     }
 
     /** Returns the number of bundles installed by this call. */
```
Comment (baldersheim) [inline container-core/src/main/java/com/yahoo/container/core/config/BundleLoader.java]: Comment is outdated

**Review by baldersheim — COMMENTED** [review]
This is unfamiliar territory. I think it is better that we go through this one together.

File: container-core/src/main/java/com/yahoo/container/core/config/BundleLoader.java (hunk 1 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/core/config/BundleLoader.java]: Thanks!

**Review by baldersheim — APPROVED** [review]
Merge at will

**Comment by gjoranv** [comment]
Awaiting merge until a system test run has successfully started.

File: jdisc_core/src/main/java/com/yahoo/jdisc/application/OsgiFramework.java [hunk 2]
```diff
@@ -60,6 +60,8 @@
     /**
      * Synchronously refresh all bundles currently loaded. Once this method returns, the
      * class loaders of all bundles will reflect on the current set of loaded bundles.
+     *
+     * NOTE: This method is no longer used by the Jdisc container framework, but kept for completeness.
```
Comment (bjorncs) [inline jdisc_core/src/main/java/com/yahoo/jdisc/application/OsgiFramework.java]: Mark as deprecated?

File: container-core/src/main/java/com/yahoo/osgi/OsgiImpl.java [hunk 3]
```diff
@@ -11,12 +11,14 @@
...
 import java.util.Collection;
 import java.util.List;
+import java.util.logging.Logger;
 
 /**
  * @author Tony Vaagenes
  * @author bratseth
  */
 public class OsgiImpl implements Osgi {
+    private static Logger log = Logger.getLogger(OsgiImpl.class.getName());
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/osgi/OsgiImpl.java]: Missing a `final`

**Review by bjorncs — APPROVED** [review]

File: jdisc_core/src/main/java/com/yahoo/jdisc/application/OsgiFramework.java (hunk 2 above)
Comment (gjoranv) [reply jdisc_core/src/main/java/com/yahoo/jdisc/application/OsgiFramework.java]: I originally did, but opted against it. It's really a natural part of the framework, and hence just a comment that it's not used by the jdisc container. It is extensively used in unit/integration tests.

**Pull request merged**